xapian-core/tests/unittest.cc

```diff
@@ -686,19 +686,29 @@ static bool test_uuid1()
 class A : public Xapian::Internal::intrusive_base {
     int x = 0;
   public:
-    A(int x_) : x(x_) {}
+    explicit A(int x_) : x(x_) {}
+
     int get_x() const {
 	return x;
     }
 };
 
 class B : public Xapian::Internal::opt_intrusive_base {
     int x = 0;
+    bool & alive;
   public:
-    B(int x_) : x(x_) {}
+    B(int x_, bool & alive_) : x(x_), alive(alive_) {
```

Comment:
I realised, initialising variable `alive` in constructor may not be the right way. It causes no problem here. If  `class B` is made to modify `alive` only in destructor, it could be clear. I assign `true` to `alive` after object  `b1` is created.  

It is trivial, but seems better.  Should I make this change?

Reply:
We do it roughly like you do elsewhere, e.g. `DestroyedFlag` in `api_none.cc` and that seems OK to me.

Why are you thinking it's better to not handle this in the class?

Also, it probably would be a good idea to delete the copy ctor and assignment operator, to make sure that we don't get a copy created implicitly (which would share the reference and the flag would get cleared when the first copy was destroyed).

Reply:
If new object got created with same destroy flag reference, tests will not be valid, like if new object sets the flag, if old object cleared it.  You can ignore this, it will not happen:)

I will delete copy constructor and assignment like you said.